First release

jsjqian opened the pull request:
Bitcamp 2017 app version 1.0

Available on both iOS and Android

Follow the README.md for install instructions

adanoff committed:
Add .gitignore (#1)

Add .gitignore (BC-001)

jdai8 committed:
Set up React Native (#9)

* Set up React Native

jdai8 committed:
Add setup docs to README (#10)

goccert25 committed:
Adds a rudimentary navigation menu tab as well as four scenes. (#3) (#11)

* Adds a rudimentary navigation menu tab as well as four scenes.

The menu tab component was found in an online library.
It fits to the platform in that for android, the menu tab will
be on top, while for iOS it will be at the bottom.
The menu tab can be found and modified inside the MenuTab.js file.
The scenes are essentially the different pages of the app. They are
found inside the scenes folder.

* Deletes a bunch of commented out lines of useless  code I neglected to delete.

* Modifies a 'require' statement to ES6 established 'import' statement.

* remove unnecessary imports and add semicolons.

jdai8 committed:
Intro page (#17)

* Add Intro scene, Aleo font

goccert25 committed:
Add Icons to Menu Tab (#15) (#18)

* Add a rudimentary event element for schedule view.

Changes are in ScheduleScene.js. Dropdown expansion has
yet to be implemented. function _onPressExpand() has no utility.

* Add trivial dropdown functionality to schedule menu.

Menu is trivial and barebones. Still needs ListView
implemented so as to allow dynamic schedule generation
based on database.

* Add relevant icon to navigation bar

Icons are different based on what platform
app is run on. There is a commented out line
to give the intro tab an icon when that is added.
New library to manage icons is also added.

* Merge fully develop and branch

Also modified RGB color values so as
to reference shared color style Jack made.

* Make coding style changes

jsjqian committed:
Add mentorship view (#16) (#19)

* Add a basic WebView for the mentorship feature

* Prepare to add features to webview

* Add a web view for the mentorship page

This includes implementing back and forward buttons. The styles will need to be updated.

* Change the header color

* Reflect the suggestions made by Jack Dai

The colors were moved into styles and the WebView string changed to a callback

* Clean up unused components and extraneous brackets

AyeAndy committed:
created new schedule view using accordion library

Commit:
modify scheme settings for iOS build

jsjqian committed:
Add Map view (#14) (#23)

* Render the map with a pdf instead

* Include the map

* Change the map orientation and refactor the MapScene

* Delete unused file

* Remove unnecessary lines and modules

* Fix style issues and unused imports

Commit:
Merge branch 'develop' of https://github.com/TerrapinFreelancers/bitcamp-app into develop

goccert25 committed:
Feature/schedule view (#24)

* Changed some fonts and spacings of the schedule

* currently working version with chevron dropdown working correctly

* fix chevron drop down arrow

* Alter .gitignore to ignore vscode files

* fix large margin at top of android, highlighting on date press in
schedule view, and changes schema of firebase to be more intuitive

* add persistent data, so without internet connection
schedule view will display latest pull

* Refactor schedule scene

Fixes android accordion animation
Removes the empty section headers warning
Combines AccordionMenu with ScheduleScene
Simplifies Accordion component by hardcoding props

jsjqian committed:
Feature/lock orientation (#25) (#28)

* Added locked portrait orientation, only tested on Android

* modify ios settings so lock orientation works with ios.

specifically adds header search paths in xcode so project can find orientation library files at compile time.

jsjqian committed:
Remove extraneous views in the mentorship view

* Add empty page for loading screen for mentor scene

* Remove extraneous features

jsjqian committed:
Create a countdown view for the front page

* Implement most of the countdown scene

Complete
- Background changes based on time of day using opacity
- Flickering flame gif
- Time remaining changes and text changes as well

Need to finish
- Change text color based on time?
- Adjust the set background values?

* Complete the countdown view, may need to clean up files and code later

* Refactor Countdown and Timer

I don't think we need to update the bg color every second.
We can just set it once.  I also cleaned up the code.

goccert25 committed:
Finish the prize view and implements notifications

* Gets the company names and puts it into the list from the JSON

* in the process of making the dynamic pages

* modified prize view

* basic list functionality

* got working list

* small modifications

* a couple of brief edits to help sruj

* fixed prize view

* prevented null click

* removed warning

* Use firebase from one account

* Update with new account

* Add push notifications and changed app icon for Android

* style CompanyInfo.js

* implement cache system for file view. Still need to clean up code and convert to ES6

* convert to ES6 and code clean up

* Change the name of the app

* implement firebase notifications for ios in connection to main firebase account

* Feature/remove cache (#32)

* Remove caching

* Created shrinkwrap json for version control

* Cache only available for iOS

jsjqian committed:
Add google-services json

jsjqian committed:
Merge branch 'develop' of https://github.com/TerrapinFreelancers/bitcamp-app into develop

jdai8 reviewed (approved):
:man_with_gua_pi_mao:

jsjqian committed:
Feature/bugfixes 1 (#34)

* Change font and make company variable

* Company and Location no longer required in schedule

* Minor style changes

* Add packages to json

srujant commented:
Hey, I was wondering if I could quickly map the back button for the Challenge View for Android. It really seems like an intuitive thing to add.

*EDIT: I added some code to the ChallengeScene that maps the back button for Android. I don't want to push it yet. I also see that if the rotation is on and I open the app in landscape mode, it breaks the layout of the whole app. 

*EDITEDIT: I pushed it to a separate branch for now (since I didn't know if it belonged in the bug fixes branch)

srujant committed:
mapped back button for android

srujant committed:
reverted to old challengescene

srujant committed:
new branch

srujant committed:
Merge pull request #36 from TerrapinFreelancers/feature/backButton

Check to see if mapping for android back button works

jdai8 committed:
Make the Countdown Scene look nicer (#38)

Add the full bitcamp logo
Remove the countdown message in the middle
Change D to Days, H to Hours, etc.

goccert25 committed:
fix accordion rendering so dropdown is smooth for components rendered off screen (#39)

jsjqian committed:
Daylight savings and Android APK (#40)

* Adjust the background for daylight savings

* Document how to make the signed APK for Android

* Update README.md

* Update README.md

* Readjusted daylight savings

* Add credit for gradle snippet

Commit:
modify ios launch screen
